omod/src/main/webapp/resources/scripts/directives/select-drug.js

```diff
@@ -36,7 +38,7 @@ angular.module('uicommons.widget.select-drug', [ 'drugService', 'ui.bootstrap' ]
                 }
             },
             template: '<input type="text" id="{{ inputId }}" ng-model="ngModel" ng-blur="verify()" ' +
-                'typeahead="drug as drug.display for drug in search($viewValue) | filter:$viewValue" ' +
```

Comment:
The function returns promise, not array. therefore, it throw Error: [filter:notarray].

Ideally, the filtering should be done on server-side and it is happening so in search function. therefore, no need to have filter in client side.
https://stackoverflow.com/a/32597661